### PR TITLE
fix: should not throw error when percent is null

### LINKED ISSUE #40377: Progress throw when providing undefined percent
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/embed/antd-reproduction-template-forked-btfox7?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce

Just provide an undefined / null to percent

### What is expected?

Not to throw (official default for percent is 0)

### What is actually happening?

It throws

| Environment | Info |
| --- | --- |
| antd | 5.1.6 |
| React | react 18 |
| System | Linux |
| Browser | all |

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #40377

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: should not throw error when percent is null |
| 🇨🇳 Chinese | fix: should not throw error when percent is null |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/progress/progress.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ const Progress: React.FC<ProgressProps> = (props) => {
   const percentNumber = React.useMemo<number>(() => {
     const successPercent = getSuccessPercent(props);
     return parseInt(
-      successPercent !== undefined ? successPercent.toString() : percent.toString(),
+      successPercent !== undefined ? (successPercent || 0).toString() : (percent || 0).toString(),
```
Comment (li-jia-nan) [inline components/progress/progress.tsx:74]: 这个在上面解构的时候，传 null 的话不走默认值，还是取 null，坑了

File: components/progress/progress.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ const Progress: React.FC<ProgressProps> = (props) => {
   const percentNumber = React.useMemo<number>(() => {
     const successPercent = getSuccessPercent(props);
     return parseInt(
-      successPercent !== undefined ? successPercent.toString() : percent.toString(),
+      successPercent !== undefined ? (successPercent ?? 0)?.toString() : (percent ?? 0)?.toString(),
```
Comment (li-jia-nan) [inline components/progress/progress.tsx:74]: 这个在上面解构的时候，传 null 的话不走默认值，还是取 null，坑了

Comment (afc163) [reply components/progress/progress.tsx:74]: 这个代码都有点看不懂了

**Review by afc163 — APPROVED** [review]

File: components/progress/progress.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/progress/progress.tsx:74]: 就是给个默认值，如果 percent 是 null 的话就取 0

**Pull request merged**